includes/discovery/sensors/dbm/mrv-optidriver.inc.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * LibreNMS Interface Power dBm module for the MRV® OptiDriver® Optical Transport Platform
+ *
+ * © 2017 Chris A. Evans <thecityofguanyu@outlook.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+echo 'MRV OptiDriver ';
+
+$multiplier = 1;
+$divisor = 1000;
+$limit_low = -100;
+$warn_limit_low = -100;
+$limit = 100;
+$warn_limit = 100;
+
+foreach ($pre_cache['mrv-optidriver_oids'] as $index => $entry) {
+    $oid = '.1.3.6.1.4.1.629.200.8.1.1.32.' . $index;
+    $descr = dbFetchCell('SELECT `ifDescr` FROM `ports` WHERE `ifName`= ? AND `device_id` = ?', array($index, $device['device_id'])) . ' Rx Power';
+    $currentrx = $entry['nbsCmmcPortRxPower'];
+    discover_sensor(
+        $valid['sensor'],
+        'dbm',
+        $device,
+        $oid,
+        'rx-' . $index,
+        'mrv-optidriver',
+        $descr,
+        $divisor,
+        $multiplier,
+        $limit_low,
+        $warn_limit_low,
+        $warn_limit,
+        $limit,
+        $currentrx,
+        'snmp'
+    );
+}
+
+foreach ($pre_cache['mrv-optidriver_oids'] as $index => $entry) {
```

Comment:
`if (is_numeric($entry['nbsCmmcPortRxPower']) {` and `if (is_numeric($entry['nbsCmmcPortTxPower']) {` is what I have come up with thus far.